docs/developer/custom-form-component-support.md

```diff
@@ -0,0 +1,42 @@
+# Custom Form Component Support
+This is an extension to the [basic form support](https://github.com/kubeapps/kubeapps/blob/master/docs/developer/basic-form-support.md#basic-form-support)
+
+## Possible use cases
+- Custom UI component that are not yet supported by the basic components (e.g: radio selectors)
+- Consuming third party APIs for component values and validation
+
+## Adding a bundle
+The js bundles are loaded as a configMap and its value can be set in the `values.yaml` file. By setting the path to the bundle in the `values.yaml` we are able to leverage the [Helm .Files object](https://helm.sh/docs/chart_template_guide/accessing_files/#glob-patterns) to load the bundle into the configMap. For example:
+```
+  customComponents: "../custom_components.js"
+```
+In this example Helm will look a directory above the `values.yaml` 
+
+## Render a custom component
+Signaling to the Kubeapps dashboard that you want to render a custom component is pretty straight forward.  Simply add a `customComponent` field to any form parameter defined in the  `values.json.schema` and that will tell the react application to fetch the component from the custom js bundle. An example parameter could look like:
+```
+    "databaseType": {
+      "type": "string",
+      "form": true,
+      "enum": ["mariadb", "postgresql"],
+      "title": "Database Type",
+      "description": "Allowed values: \"mariadb\" and \"postgresql\"",
+      "customComponent": {
+          "type": "radio",
+          "className": "primary-radio"
+      }
+    }
+```
+Note: The `customComponent` field **MUST BE AN OBJECT**. This design decision was made so that developers can pass extra values/properties into their custom components.
```

Comment:
is there any mandatory fields in that object? (I assume `className` is required?)

Reply:
I think its up to the person integrating whether they want/need to pass anything extra to the components. If they are only rendering one component then it might be easier for them to just hardcode the className in the component they are importing